base/keylist/keylist.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2024, Cogent Core. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/*
+package keylist implements an ordered list (slice) of items,
+with a map from a key (e.g., names) to indexes,
+to support fast lookup by name.
+This is a different implementation of the [ordmap] package,
+that has separate slices for Values and Keys, instead of
+using a tuple list of both. The awkwardness of value access
+through the tuple is the major problem with ordmap.
+*/
+package keylist
+
+import (
+	"fmt"
+	"slices"
+)
+
+// TODO: probably want to consolidate ordmap and keylist
+
+// List implements an ordered list (slice) of Values,
+// with a map from a key (e.g., names) to indexes,
+// to support fast lookup by name.
+type List[K comparable, V any] struct { //types:add
+	// List is the ordered slice of items.
+	Values []V
+
+	Keys []K
+
+	// indexes is the key-to-index mapping.
+	indexes map[K]int
+}
+
+// New returns a new Key List.  Zero value
+// is usable without initialization, so this is
+// just a simple standard convenience method.
+func New[K comparable, V any]() *List[K, V] {
+	return &List[K, V]{}
+}
+
+func (kl *List[K, V]) newIndexes() {
```

Comment:
You should probably call this `makeIndexes` to be consistent with the `make` function that you are calling.